lib/puppet/provider/file_line/ruby.rb

```diff
@@ -34,13 +34,22 @@ def lines
 
   def handle_create_with_match()
     regex = resource[:match] ? Regexp.new(resource[:match]) : nil
+    regex_after = resource[:after] ? Regexp.new(resource[:after]) : nil
     match_count = count_matches(regex)
+
     if match_count > 1 && resource[:multiple].to_s != 'true'
      raise Puppet::Error, "More than one line in file '#{resource[:path]}' matches pattern '#{resource[:match]}'"
     end
+
     File.open(resource[:path], 'w') do |fh|
       lines.each do |l|
         fh.puts(regex.match(l) ? resource[:line] : l)
+        if (match_count == 0 and resource[:after])
```

Comment:
They would both logically be the same, nil or ! nil, just a matter of readability and preference I guess.